### PR TITLE
feature(generator): add utility functions to handle non LRO services

### PR DESCRIPTION
**Pull request by scotthart**
Adding support for detecting which services have longrunning operation methods and which do not. 
Modifying how includes are emitted to handle empty strings.
This PR is a chunk of a larger imminent PR that I could carve off.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5604)

### CODE REVIEW

File: generator/internal/codegen_utils.cc [hunk 1]
```diff
@@ -28,11 +28,17 @@ namespace generator_internal {
 std::string GeneratedFileSuffix() { return ".gcpcxx.pb"; }
 
 std::string LocalInclude(absl::string_view header) {
-  return absl::StrCat("#include \"", header, "\"\n");
+  if (header.length() > 0) {
```
Comment (devbww) [inline generator/internal/codegen_utils.cc:31]: Prefer `!header.empty()`?

And perhaps invert the test:
```
  if (header.empty()) return {};
  return absl::StrCat("#include \"", header, "\"\n");
```

File: generator/internal/codegen_utils.cc [hunk 2]
```diff
@@ -28,11 +28,17 @@ namespace generator_internal {
...
+    return absl::StrCat("#include \"", header, "\"\n");
+  }
+  return {};
 }
 
 std::string SystemInclude(absl::string_view header) {
-  return absl::StrCat("#include <", header, ">\n");
+  if (header.length() > 0) {
```
Comment (devbww) [inline generator/internal/codegen_utils.cc:38]: Ditto?

File: generator/internal/codegen_utils_test.cc [hunk 3]
```diff
@@ -29,10 +29,20 @@ TEST(LocalInclude, Success) {
             LocalInclude("google/cloud/status.h"));
 }
 
+TEST(LocalInclude, Empty) {
+  std::string empty;
```
Comment (devbww) [inline generator/internal/codegen_utils_test.cc:33]: Nit: Whenever I have a variable whose name asserts something about its value, I tend to make it `const`.

File: generator/internal/predicate_utils.cc [hunk 4]
```diff
@@ -27,6 +27,17 @@ using ::google::protobuf::Descriptor;
...
+  bool has_longrunning_method = false;
+  for (int i = 0; !has_longrunning_method && i < service.method_count(); ++i) {
+    if (service.method(i)->output_type()->full_name() ==
+        "google.longrunning.Operation") {
+      has_longrunning_method = true;
+    }
+  }
+  return has_longrunning_method;
+}
+
```
Comment (devbww) [inline generator/internal/predicate_utils.cc:40]: Nit: Consider avoiding the variable with judicious placement of `return true` and `return false`.

**Review by devbww — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5604)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @devbww)

___
*[generator/internal/codegen_utils.cc, line 31 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5604#-MO9YoC1XCFzSn9CnSM8-r1-31:-MOCnLcd4HOrImwkSNU2:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/154527f46c617f6be3ee19bb1f9f82b5eb7b1340/generator/internal/codegen_utils.cc#L31)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Prefer `!header.empty()`?

And perhaps invert the test:
```
  if (header.empty()) return {};
  return absl::StrCat("#include \"", header, "\"\n");
```
</blockquote></details>

Done.
___
*[generator/internal/codegen_utils.cc, line 38 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5604#-MO9YoC1XCFzSn9CnSM8-r1-38:-MOCnMJoDBpHBAm2RoIu:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/154527f46c617f6be3ee19bb1f9f82b5eb7b1340/generator/internal/codegen_utils.cc#L38)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Ditto?
</blockquote></details>

Done.

**Pull request merged**